test/db/src/main/java/org/commonjava/indy/core/data/GroupDataManagerTCK.java

```diff
@@ -243,4 +246,42 @@ public int compare( final Group g1, final Group g2 )
         assertThat( g.getName(), equalTo( grp2.getName() ) );
     }
 
+    @Test
+    public void createTwoGroupsAndAffectedByForOneLevel() throws Exception{
+        final StoreDataManager manager = getFixtureProvider().getDataManager();
+        final StoreKey central = new StoreKey( StoreType.remote, "central" );
+        final StoreKey repo2 = new StoreKey( StoreType.remote, "repo2" );
+        Group g1 = new Group( "g1", central );
+        Group g2 = new Group( "g2", repo2 );
+
+        store( g1, g2 );
+
+        List<StoreKey> keys = Arrays.asList( central, repo2 );
+
+        Set<StoreKey> gKeys =
+                manager.getGroupsAffectedBy( keys ).stream().map( Group::getKey ).collect( Collectors.toSet() );
+
+        assertThat( gKeys.contains( g1.getKey() ), equalTo( Boolean.TRUE ) );
+        assertThat( gKeys.contains( g2.getKey() ), equalTo( Boolean.TRUE ) );
+    }
+
+    @Test
+    public void createTwoGroupsAndAffectedByForTwoLevel() throws Exception{
+        final StoreDataManager manager = getFixtureProvider().getDataManager();
+        final StoreKey central = new StoreKey( StoreType.remote, "central" );
+        Group g1 = new Group( "g1", central );
+        Group g2 = new Group( "g2", g1.getKey() );
+
+        store(g1, g2);
+
+        List<StoreKey> keys = Collections.singletonList( central );
+
+        Set<StoreKey> gKeys =
+                manager.getGroupsAffectedBy( keys ).stream().map( Group::getKey ).collect( Collectors.toSet() );
+
+        assertThat( gKeys.contains( g1.getKey() ), equalTo( Boolean.TRUE ) );
+        //FIXME: should this group:g2 be in result?
```

Comment:
Yes, this is exactly right. The idea is to traverse a sort of reverse mapping of the groups that directly or indirectly contain some key.